sdk-tools/src/sign.rs

```diff
@@ -0,0 +1,272 @@
+    /// and write it to the given output data path.
+    pub fn gendata(
+        &mut self,
+        unsigned_enclave: &Path,
+        config_path: &Path,
+        output_datfile: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("gendata")
+            .arg("-enclave")
+            .arg(unsigned_enclave)
+            .arg("-config")
+            .arg(config_path)
+            .arg("-out")
+            .arg(output_datfile);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        if self.resign {
+            cmd.arg("-resign");
+        }
+
+        cmd
+    }
+
+    /// Combine an unsigned enclave and signature into the output enclave, after
+    /// checking the signature.
+    pub fn catsig(
+        &mut self,
+        unsigned_enclave: &Path,
+        config_path: &Path,
+        public_key_pem: &Path,
+        gendata_output: &Path,
+        signature: &Path,
+        output_enclave: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("catsig")
+            .arg("-enclave")
+            .arg(unsigned_enclave)
+            .arg("-config")
+            .arg(config_path)
+            .arg("-key")
+            .arg(public_key_pem)
+            .arg("-unsigned")
+            .arg(gendata_output)
+            .arg("-sig")
+            .arg(signature)
+            .arg("-out")
+            .arg(output_enclave);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        cmd
+    }
+
+    /// Examine a signed enclave file and dump the data
+    pub fn dump(
+        &mut self,
+        signed_enclave: &Path,
+        css_file_path: &Path,
+        dump_file_path: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("dump")
+            .arg("-enclave")
+            .arg(signed_enclave)
+            .arg("-dumpfile")
+            .arg(dump_file_path)
+            .arg("-cssfile")
+            .arg(css_file_path);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        if self.resign {
+            cmd.arg("-resign");
+        }
+
+        cmd
+    }
+}
+
+/// Construct a new SgxSign utility around the given executable path
+impl From<PathBuf> for SgxSign {
+    fn from(sgx_sign_path: PathBuf) -> Self {
+        Self {
+            sgx_sign_path,
+            ignore_relocation_error: false,
+            ignore_init_section_error: false,
+            resign: false,
+        }
+    }
+}
```

Comment:
:wrench: I think this should change not sure exactly how:
1. Instead of being the location of `sgx_sign` this should be the location of the `enclave` file being signed and a builder method to add the config
2. Remove this from and have a `new()` that takes an enclave and a config file.  The config file is optional according to SGX sign, but it looks like the methods are all requiring it.  `dump` is an exception here but I'm thinking `dump` doesn't belong in `SgxSign` and it should be in a separate struct.
3. Add builder methods to add the input enclave file and config file.

I'm leaning toward 1 to match `Edger8r`. I'm not opposed to rethinking `Edger8r`
My reasoning is that I'm thinking of inputs -> outputs. "What does the SgxSign" operate on?
```mermaid
graph LR
    A[enclave] --> B(sgx_sign)
    C[config] --> B
    B --> D[signed enclave]
```
The path to the signing binary doesn't seem needed, all of the building logic needs the path to the SGX_SDK so that already had to be set up by clients and setting it up per tool seem unnecessary at this time.

Reply:
The original purpose was to just wrap the execution of the `sgx_sign` binary, without being opinionated about what you're using it for.

I'm also fine to remove `Default` (and therefore the dependency on `core-build`) and have `From<impl AsRef<Path>>` be the only constructor.

Reply:
Will be fixed in https://github.com/mobilecoinfoundation/sgx/issues/264

Reply:
https://github.com/mobilecoinfoundation/sgx/issues/264